tests/cid.rs

```diff
@@ -82,6 +82,21 @@ fn test_cid_not_as_bytes() {
         .expect_err("shouldn't have parsed a tagged CID as a byte array");
     from_slice::<serde_bytes::ByteBuf>(&cbor_cid[2..])
         .expect("should have parsed an untagged CID as a byte array");
+
+    #[cfg(feature = "no-cid-as-bytes")]
+    {
+        #[derive(Debug, Deserialize)]
+        struct NewType(ByteBuf);
+
+        #[derive(Debug, Deserialize)]
+        #[serde(untagged)]
+        enum BytesInEnum {
+            MyCid(NewType),
+        }
```

Comment:
What I'm interested in is what happens when you do this without the feature. Can you add a test for that with these types and run the assertions as expected but also document that this behaviour may be undesirable? This is the key point of the issue right here, isn't it?

Reply:
@rvagg sorry for the delay, I've pushed a new commit which should address your concerns. If not, please let me know.